internal/engine/synclet_build_and_deployer.go

```diff
@@ -126,10 +160,11 @@ func (sbd *SyncletBuildAndDeployer) updateViaSynclet(ctx context.Context,
 	// TODO(maia): can refactor MissingLocalPaths to just return ContainerPaths?
 	containerPathsToRm := build.PathMappingsToContainerPaths(toRemove)
 
-	deployInfo, ok := sbd.getDeployInfoForImage(state.LastResult.Image)
-	if !ok {
+	deployInfo, ok := sbd.deployInfoForImageBlocking(ctx, state.LastResult.Image)
+	if !ok || deployInfo == nil {
```

Comment:
Is it possible for ok to be true and deployInfo to be nil, or does that mean we have a bug somewhere? If the former, that seems weird. If the latter, I might prefer to leave the `|| deployInfo == nil` off, so that we get a panic instead of conflating one bug (we're putting nils in there somehow) with another (we tried to get the container id and failed)

Reply:
i think the latter, good call